cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -289,10 +290,16 @@ func createCoreDNSAddon(deploymentBytes, serviceBytes, configBytes []byte, clien
 				return err
 			}
 		}
-	} else {
+	} else if isCoreDNSCorefileEmptyOrDefault(corefile) {
+		// If the Corefile is empty or default, the latest default Corefile will be applied
```

Comment:
I think the only time we get in here is when the default corefile is empty or already identical to the new default corefile (no migrations required).